test/functional/wallet_upgradetohd.py

```diff
@@ -190,8 +190,8 @@ def run_test(self):
         node.stop()
         node.wait_until_stopped()
         self.start_node(0, extra_args=['-rescan'])
-        assert_raises_rpc_error(-13, "Error: Please enter the wallet passphrase with walletpassphrase first.", node.upgradetohd, mnemonic)
-        assert_raises_rpc_error(-14, "The wallet passphrase entered was incorrect", node.upgradetohd, mnemonic, "", "wrongpass")
+        assert_raises_rpc_error(-4, "Wallet encrypted but passphrase not supplied. Please refer to \"help upgradetohd\" for guidance.", node.upgradetohd, mnemonic)
+        assert_raises_rpc_error(-1, "GenerateNewHDChain: supplied incorrect passphrase", node.upgradetohd, mnemonic, "", "wrongpass")
```

Comment:
also all over all RPC the check `EnsureWalletIsUnlocked` is a general-use-practice, that is used for _all_ other rpc which requires wallet to be unlocked, and don't really care how exactly wallet is unlocked - because it is a wrong passphrase or because it is not provided

Reply:
`EnsureWalletIsUnlocked()` doesn't tell if you have the wrong passphrase ([source](https://github.com/dashpay/dash/blob/1394c41c8d0afb8370726488a2888be30d238148/src/wallet/rpcwallet.cpp#L127-L132)), it tells you to use `walletpassphrase` (the RPC) if your wallet isn't unlocked. 

As I mentioned in my PR description, that reenforces the idea that this is just like any other privileged command, unlocking your wallet let you perform the privileged operation, when this isn't the case. Having an unlocked wallet won't give us the passphrase needed so the unlock state doesn't matter. Under the hood, we're locking-unlocking the wallet anyways.

The incorrect passphrase error comes from within `upgradetohd()` ([source](https://github.com/dashpay/dash/blob/1394c41c8d0afb8370726488a2888be30d238148/src/wallet/rpcwallet.cpp#L2808-L2813)) and I had thought about doing the passphrase validation in `upgradetohd` to keep around the friendlier error message and retain the error code but that would require doing the `Lock()`-`Unlock()` routine, only for `GenerateNewHDChain()` to do it again (and I'd prefer to keep it in `GenerateNewHDChain()` than only in `upgradetohd()`).

We could make the message more friendly by changing the text in `GenerateNewHDChain()` itself but that would mean bucking with error message convention for exceptions thrown from non-RPC function bodies of "[function name]: [error message]", thoughts?